src/routes/things.js

```diff
@@ -47,8 +56,74 @@ let router = express.Router();
  * }
  **/
 
-router.get('/Things', (req, res) => {
-  res.status(200).send();
+const resource = 'Things';
+
+router.get('/', (req, res) => {
+  db().then((models) => {
+    if (req.params && req.params[0]) {
+      models.Things.findById(req.params[0]).then(thing => {
+        res.status(200).send(response.generate(thing));
+      });
+    } else {
+      models.Things.findAll().then(things => {
+        res.status(200).send(response.generate(things));
+      });
+    }
+  })
+  .catch(() => {
+    ApiError(res, 500, ERRNO_INTERNAL_ERROR, INTERNAL_ERROR);
+  });
+});
+
+router.post('/', (req, res) => {
+  db().then((models) => {
+    models.Things.create(req.body)
+    .then(createdThing => {
+      res.location('/' + resource + '/' + createdThing.id);
+      res.status(201).send();
+    })
+    .catch(() => {
+      res.status(400).send();
+    });
+  });
+});
+
+router.patch('/', (req, res) => {
+  if (!req.params || !req.params[0]) {
+    ApiError(res, 404, ERRNO_RESOURCE_NOT_FOUND, BAD_REQUEST);
+
+    return;
+  }
+
+  db().then((models) => {
+    models.Things.findById(req.params[0]).then(thing => {
+      thing.update(req.body).then(updatedThing => {
+        res.status(200).send(response.generate(updatedThing));
+      });
+    });
+  });
+});
+
+router.delete('/', (req, res) => {
+  if (!req.params || !req.params[0]) {
+    ApiError(res, 404, ERRNO_RESOURCE_NOT_FOUND, BAD_REQUEST);
+
+    return;
+  }
+
+  db().then((models) => {
+    models.Things.findById(req.params[0]).then(thing => {
+      thing.destroy().then(() => {
```

Comment:
We need to remove all the datastreams for this thing http://docs.opengeospatial.org/is/15-078r6/15-078r6.html#table_25

Add a new `// XXX` plus issue, please.